generators/csharp/sdk/src/oauth/OauthTokenProviderGenerator.ts

```diff
@@ -0,0 +1,273 @@
+            name: OauthTokenProviderGenerator.GET_ACCESS_TOKEN_ASYNC_METHOD_NAME,
+            parameters: [],
+            body: this.getAccessTokenBody(),
+            return_: csharp.Type.string()
+        });
+    }
+
+    private getAccessTokenBody(): csharp.CodeBlock {
+        const expiresIn = this.getExpiresIn();
+        const tokenEndpoint = this.scheme.configuration.tokenEndpoint;
+
+        return csharp.codeblock((writer) => {
+            writer.controlFlow(
+                "if",
+                csharp.codeblock((writer) => {
+                    writer.write(`${OauthTokenProviderGenerator.ACCESS_TOKEN_FIELD.name} == null`);
+                    // check expiresIn if present in the IR
+                    if (expiresIn != null) {
+                        writer.write(`|| DateTime.UtcNow >= ${OauthTokenProviderGenerator.EXPIRES_AT_FIELD.name}`);
+                    }
+                })
+            );
+
+            writer.writeNodeStatement(
+                csharp.codeblock((writer) => {
+                    const requestType = this.tokenEndpoint.requestBody?._visit({
+                        reference: (value) => {
+                            return this.context.csharpTypeMapper.convert({ reference: value.requestBodyType });
+                        },
+                        inlinedRequestBody: (value) => {
+                            return csharp.Type.reference(
+                                this.context.getRequestWrapperReference(
+                                    this.tokenEndpointReference.serviceId,
+                                    value.name
+                                )
+                            );
+                        },
+                        fileUpload: (value) => undefined,
+                        bytes: (value) => undefined,
+                        _other: (value) => undefined
+                    });
+                    if (requestType == null || requestType.internalType.type !== "reference") {
+                        throw new Error("Failed to get request class reference");
+                    }
+
+                    writer.write("var tokenResponse = ");
+                    writer.writeNode(
+                        csharp.invokeMethod({
+                            async: true,
+                            method: `client.${this.tokenEndpoint.name.pascalCase.safeName}`,
+                            arguments_: [
+                                csharp.instantiateClass({
+                                    classReference: requestType.internalType.value,
+                                    // TODO(dsinghvi): assumes only top level client id and client secret inputs
+                                    arguments_: [
+                                        {
+                                            name: this.context.getNameForField(
+                                                this.scheme.configuration.tokenEndpoint.requestProperties.clientSecret
```

Comment:
I think this should be the following:
```suggestion
                                                this.scheme.configuration.tokenEndpoint.requestProperties.clientId
```